link/src/main/java/com/stripe/android/link/ui/signup/Model.kt

```diff
@@ -0,0 +1,18 @@
+package com.stripe.android.link.ui.signup
+
+data class SignUpViewState(
+    val loading: Boolean = false
+)
+
+sealed interface SignUpAction {
+    data object SignUpClicked : SignUpAction
+}
+
+sealed interface SignUpResult {
+    data object ShowLoader : SignUpResult
+    data class SendEffect(val effect: SignUpEffect) : SignUpResult
```

Comment:
I think having a result called `SendEffect` is conflating results with effects. It makes more sense to me for this to be something like `SignUpSucceeded` or `SignUpFailed`. Then the `resultToEffect` function would be responsible for determining that e.g. if sign up succeeds, then we should navigate to the wallet screen